Assets/PatchKit/Securing by the license key/SecuringByLicenseKey.cs

```diff
@@ -1,204 +1,312 @@
 ﻿using System;
-using System.Collections;
 using System.Collections.Generic;
-using System.ComponentModel;
 using System.IO;
 using System.Net;
-using UnityEngine;
 using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
 using UnityEditor;
+using UnityEngine;
+using UnityEngine.SceneManagement;
 
-public class SecuringByLicenseKey : MonoBehaviour
+namespace Assets.PatchKit.Securing_by_the_license_key
 {
-    private const string AppDataFileName = "app_data.json";
-    private static bool isFirstStart = true;
-    private string _appDataPath;
-    private string _filePath;
-    private GUIStyle guiStyle;
-    private Rect windowRect = new Rect((Screen.width - 300) / 2, (Screen.height - 100) / 2, 300, 100);
-    private bool show = false;
-    private string _labelMessage;
-    private string _contentMessage;
-    private string host = "keys2.patchkit.net";
-    private string path = "/v2/keys/{key}";
-    
-    public string AppSecret;
-
-    /// <summary>
-    /// Data structure stored in file.
-    /// </summary>
-    private struct Data
+    public class SecuringByLicenseKey : MonoBehaviour
     {
-        [DefaultValue("patcher_data")] [JsonProperty("file_id", DefaultValueHandling = DefaultValueHandling.Populate)]
-        public string FileId;
-
-        [DefaultValue("1.0")] [JsonProperty("version", DefaultValueHandling = DefaultValueHandling.Populate)]
-        public string Version;
-
-        [DefaultValue("")] [JsonProperty("product_key", DefaultValueHandling = DefaultValueHandling.Populate)]
-        public string ProductKey;
+        private const string AppDataFileName = "app_data.json";
+        private static bool _isFirstStart = true;
+        private string _appDataPath;
+        private string _filePath;
+        private GUIStyle _guiStyle;
+        private Rect _windowRect = new Rect((Screen.width - 300) / 2, (Screen.height - 160) / 2, 300, 160);
+        private bool _show = false;
+        private bool _notFoundKey = false;
+        private string _labelMessage;
+        private string _contentMessage;
+        private string _tmpKey;
+        private int _idWindow = 0;
+        private string _host = "keys2.patchkit.net";
+        private string _path = "/v2/keys/{key}";
+        private string _productKey = "product_key";
+        
+        public Options Option;
+        public List<GameObject> Objects;
+        public SceneAsset SceneLoad;
+        public bool NotExecuteInEditor = false;
+        public string AppSecret;
+        private JObject _data;
```

Comment:
Is there an equivalent for `JObject` in `UnityEngine.JsonUtility`?

If so, we can switch the usage from Newtonsoft.Json as in editior extension.

Reply:
done